src/code/z_message.c

```diff
@@ -223,339 +807,4300 @@ void Message_FindMessage(PlayState* play, u16 textId) {
+u16 D_801D027C[] = { 0x2001, 0x2003, 0x2004, 0x2002 };
+
+// Counterpart to NES sMaskCodeTextENG
+u16 D_801D0284[] = { 0x90D4, 0x90C2, 0x89A9, 0x97CE };
+
+#ifdef NON_MATCHING
+// https://decomp.me/scratch/ZTbP4
+void Message_Decode(PlayState* play) {
+    MessageContext* msgCtx = &play->msgCtx;
+    Font* font = &msgCtx->font; // spF0
+    Player* player;             // spEC
+    s16 decodedBufPos;          // spEA
+    u32 timeToMoonCrash;
+    s16 temp2;
+    s16 spE0;
+    s32 pad4;
+    s16 digits[4]; // spD4/D6/D8/DA
+    s16 spD2;
+    f32 timeInSeconds;
+    s32 charTexIdx; // spC8
+    s16 var_v0;
+    s16 loadChar;
+    f32 spC0;
+    f32 var_fs0;
+    s16 index;
+    s16 value;
+    s16 playerNameLen;
+    s16 i; // s1
+    s16 spAC[4];
+    u8 var_s3_8;
+    u16 curChar;
+    u32 temp;
+    u8* ptr2;
+
+    player = GET_PLAYER(play);
+    msgCtx->textDelayTimer = 0;
+    msgCtx->textDelay = msgCtx->textDelayTimer;
+    msgCtx->textFade = 0;
+    spC0 = 0.0f;
+    font->unk_11D88 = (font->unk_11D88 ^ 1) & 1;
+
+    if ((gSaveContext.options.language == 0) && !msgCtx->textIsCredits) {
+        spD2 = 0;
+        spE0 = 0;
+        decodedBufPos = 0;
+        charTexIdx = 0;
+        Message_SetupLoadItemIcon(play);
+
+        while (true) {
+            curChar = msgCtx->decodedBuffer.wchar[decodedBufPos] = font->msgBuf.wchar[msgCtx->msgBufPos];
+
+            if ((curChar == 9) || (curChar == 0xB) || (curChar == 0x110) || (curChar == 0x111) || (curChar == 0x112) ||
+                (curChar == 0x103) || (curChar == 0x230) || (curChar == 0x500) || (curChar == 0x240) ||
+                (curChar == 0x104)) {
+
+                msgCtx->msgMode = MSGMODE_TEXT_DISPLAYING;
+                msgCtx->textDrawPos = 1;
+                msgCtx->unk11FFA = msgCtx->textboxY + 6;
+                msgCtx->unk11F1A[spD2] = 0;
+                if (msgCtx->unk11F18 == 0) {
+                    msgCtx->unk11F1A[spD2] = (s16)((msgCtx->textCharScale * 16.0f * 16.0f) - spC0) / 2;
+                }
+                spC0 = 0.0f;
+                if (curChar == 0xB) {
+                    if ((msgCtx->textBoxType != TEXTBOX_TYPE_3) && (msgCtx->textBoxType != TEXTBOX_TYPE_4)) {
+                        if (spE0 < 2) {
+                            msgCtx->unk11FFA = msgCtx->textboxY + 0x16;
+                        } else if (spE0 == 2) {
+                            msgCtx->unk11FFA = msgCtx->textboxY + 0xE;
+                        }
+                    }
+                } else {
+                    if ((msgCtx->textBoxType != TEXTBOX_TYPE_3) && (msgCtx->textBoxType != TEXTBOX_TYPE_4)) {
+                        if (spE0 == 0) {
+                            msgCtx->unk11FFA = msgCtx->textboxY + 0x16;
+                        } else if (spE0 == 1) {
+                            msgCtx->unk11FFA = msgCtx->textboxY + 0xE;
+                        }
+                    }
+                }
+
+                {
+                    s32 pad;
```

Comment:
Is this necessary to prevent some swaps? We usually call those `requiredScopeTemp`. If it's just to pad the stack out, then ignore this.

Reply:
Since it is NON_MATCHING, I think I'd prefer leaving it as is since we aren't 100% sure it will need to stay.

Reply:
Ah, sorry, I missed entirely that this was in a NON_MATCHING function! Yeah, just ignore this feedback then.